operator/v1/types.go

```diff
@@ -63,22 +63,22 @@ type HubCoreList struct {
 // +kubebuilder:subresource:status
 // +kubebuilder:resource:scope=Cluster
 
-// SpokeCore represents a deployment of nucleus core agent on spoke cluster.
+// Klusterlet represents a deployment of operator core agent on spoke cluster.
```

Comment:
I think it would be `Klusterlet represents a deployment of core agents on spoke cluster`?

Reply:
I love the name Klusterlet, but it's currently in use elsewhere in the system - are we ready to cut over the other APIs this would touch?

Reply:
I'm thinking of the group of agents that are currently shipped bundled with the analog of the registration agent.

Reply:
Yes, part of this is acknolwedging that the role of our "klusterlet" has mainly been about:

1. registration
2. work distribution
3. agent extensibility

Here, I think we preserve the name "Klusterlet" for functions 1&2. It will be part of a new API group and we will have a reaction in the current "Klusterlet/KlusterletConfig", but @TheRealHaoLiu is already making changes there based on adopting the new registration and `ManifestWork`.

Reply:
> Yes, part of this is acknolwedging that the role of our "klusterlet" has mainly been about:
> 
> 1. registration
> 2. work distribution
> 3. agent extensibility
> 
> Here, I think we preserve the name "Klusterlet" for functions 1&2. It will be part of a new API group and we will have a reaction in the current "Klusterlet/KlusterletConfig", but @TheRealHaoLiu is already making changes there based on adopting the new registration and `ManifestWork`.

I like this name and sentiment. I think the doc here https://github.com/open-cluster-management/api/pull/28/files#diff-d26475bf8fc8b8ed51bb729174e4f140R17 reinforces it.